admin_ui/src/components/AddRowForm.vue

```diff
@@ -67,18 +69,26 @@ export default {
                 })
             } catch (error) {
                 const data = error.response.data
+                const databaseError = data["db_error"]
+                const validationError = data["detail"]
+                let errorsArray = []
+
+                if (databaseError) {
+                    errorsArray.push(databaseError)
+                } else {
+                    validationError.forEach((item: any) => {
+                        errorsArray.push(`Field ${item.loc[1]} - ${item.msg}`)
+                    })
+                }
```

Comment:
We need to be careful here, as we can get other types of error back.

For example, if it's a 500 error we just get a string back saying '500 error'.

<img width="478" alt="Screenshot 2022-09-08 at 15 24 51" src="https://user-images.githubusercontent.com/350976/189147840-df681759-0044-4dda-aa5a-869248a4a3c7.png">

<img width="701" alt="Screenshot 2022-09-08 at 15 24 35" src="https://user-images.githubusercontent.com/350976/189147988-ff9d0894-43d9-455e-aad2-066fc9fea5f5.png">

I think it should check status codes.

 * Pydantic validation errors return a 422.
 * Database unique errors are also returned as 422.

If the status code is something else, we should just render what ever the response was.

```javascript
const data = error.response.data

let errorsArray: string[] = []

if (error.response.status == 422) {
    const databaseError = data["db_error"]
    const validationError = data["detail"]

    if (databaseError) {
        errorsArray.push(databaseError)
    } else {
        validationError.forEach((item: any) => {
            errorsArray.push(`Field ${item.loc[1]} - ${item.msg}`)
        })
    }
} else {
    errorsArray.push(data)
}
```

Reply:
We can also get a 405 status code when running in read only mode.

<img width="1218" alt="Screenshot 2022-09-08 at 15 52 02" src="https://user-images.githubusercontent.com/350976/189154662-e319d72e-df6c-417f-8112-c847fd6033f7.png">